src/libraries/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj

```diff
@@ -25,7 +25,7 @@ System.Diagnostics.DiagnosticSource</PackageDescription>
 
   <ItemGroup>
     <ILLinkSubstitutionsXmls Include="ILLink/ILLink.Substitutions.Shared.xml" />
-    <Content Include="ILLink\ILLink.Descriptors.LibraryBuild.xml" />
+    <None Include="ILLink\ILLink.Descriptors.LibraryBuild.xml" />
```

Comment:
Assuming this is only required for our own build and doesn't need to somehow be bundled (as embedded resource, or file adjacent to lib) for use in end-user consumption of our library in linker scenarios.

Reply:
@noahfalk can you please answer this?  I chatted with @tarekgh and we weren't sure about the scenario where someone uses a 10.0 package in a `net9.0` app.  In that case CoreLib would be calling into the package version of this.  Shouldn't that case also require the linker to preserve the App's copy of `MetricsEventSource.GetInstance` in which case we should just merge this with the other file, or put it in a file specific to net10 and later?

Depending on your answer here we can follow up with work in 10.0 -- it shouldn't impact this servicing PR.

Reply:
> Shouldn't that case also require the linker to preserve the App's copy of MetricsEventSource.GetInstance in which case we should just merge this with the other file

If you are trimming .NET 9 app with 10.0 version of this package, the trimmer will see the reference to this method from CoreLib and preserve it.

This file is only needed for isolated assembly trimming done as part of the dotnet/runtime repo build. The linker does not see the universe in this case, it does not see that this method is referenced via private reflection from CoreLib, and this file is needed to instruct the linker to keep the method around.

Reply:
> trimmer will see the reference to this method from CoreLib

Do we support trimming a framework dependent application?  Will the linker always get a copy of CoreLib?

Reply:
Oh - I see - we don't support this https://github.com/dotnet/linker/issues/2139